File: Server/services/models/operation.model.js
const { Schema, model } = require("mongoose");
const { ObjectId } = require("mongodb");


const operationSchema = Schema({
	localId: { type: Number, required: true },
	subject: { type: ObjectId, ref: "user", required: true },
	title: { type: String, required: true },
	account: { type: ObjectId, ref: "account", required: true },
	amount: { type: Number, required: true },
	active: { type: Boolean, required: true },
	date: { type: Date, required: true },
	description: { type: String },
	category: { type: String, default: null },
	favorite: { type: Boolean, default: false },
	imgUrl: { type: String },
});

exports.OperationModel = model("operation", operationSchema);

File: Proyect_app/app/src/main/java/com/example/proyecto_final_apps/data/local/entity/OperationModel.kt
package com.example.proyecto_final_apps.data.local.entity

import android.content.Context
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.proyecto_final_apps.data.Category
import com.example.proyecto_final_apps.data.CategoryModel
import com.example.proyecto_final_apps.ui.adapters.OperationItem
import java.util.*

@Entity
data class OperationModel(
    @PrimaryKey(autoGenerate = true)
    var localId: Int? = null,
    var accountRemoteId: String?,
    var accountLocalId:Int,
    var active: Boolean,
    var amount: Double,
    var category: Int,
    var favorite: Boolean,
    var date: String,
    val remoteId: String? = null,
    val subject: String? = null,
    var title: String,
    var description: String?,
    var requiresUpdate: Boolean? = false,
    var imgUrl: String?,
    var deletionPending:Boolean = false
)

fun OperationModel.getCategory(context: Context):CategoryModel?{
    return Category(context).getCategory(this.category);
}

fun OperationModel.toOperationItem(context: Context):OperationItem{
    return OperationItem(
        localId = localId!!,
        remoteId = remoteId,
        title = title,
        category= getCategory(context),
        amount = amount,
        active = active,
        imgUrl = imgUrl
    )
}